### ISSUE TITLE
Build oVirt Node 9 based on AlmaLinux OS 9

### ISSUE THREAD

**Issue by LKHN** [oVirt/ovirt-node-ng-image#49]
Hey oVirt Team,

Since you are considering producing oVirt Node builds based on a more stable distribution than CentOS Stream 9. As a member of the AlmaLinux OS Virt SIG, I'd like to add another option to this decision:

The AlmaLinux OS 9.0 Beta 1 is [available](https://almalinux.org/blog/almalinux-9-beta-now-available/) with its cloud images from 18th April 2022. We also expect the 9.0 stable in a couple of days. As AlmaLinux OS Virt SIG, we would love and be honoured to help.

**Comment by sandrobonazzola** [comment]
welcome aboard!

**Comment by srbala** [comment]
@sandrobonazzola thanks for the invite.

**Comment by sandrobonazzola** [comment]
Test build: https://resources.ovirt.org/repos/testing/alma/

**Comment by srbala** [comment]
@sandrobonazzola Can this ISO can work in VM environment? ISO booted ok, after install system did not booted, showed Grub shell as below. VM had 4 cpus, 8G RAM and 200GB hdd
![image](https://user-images.githubusercontent.com/1273137/171977739-86315557-e7cb-4241-8d4b-a54c19d83e81.png)

**Comment by srbala** [comment]
VM had UEFI and secure boot enabled

**Comment by sandrobonazzola** [comment]
Looks like it requires some debugging :-/ Worked on CentOS Stream 9 to this point so we need to figure out why this is happening.

**Comment by srbala** [comment]
@andrewlukoshko recommended to check EFI boot folder creation, since Almalinux and Rocky has it own https://github.com/rhinstaller/anaconda/blob/rhel-9/data/product.d/almalinux.conf#L13

**Comment by sandrobonazzola** [comment]
Thanks @srbala and @andrewlukoshko , need to dig into this.

**Comment by srbala** [comment]
@sandrobonazzola It looks like os installer creates the `grub.cfg` file under `centos` folder, it is expected under `almalinux` folder. Moving the file to right location get the system get booted.

### Current
![image](https://user-images.githubusercontent.com/1273137/172949822-80e9a544-9197-4637-a38e-69b5ab4fe61a.png)

### Expected
![image](https://user-images.githubusercontent.com/1273137/172949884-0c19221f-fa78-48b9-b6cd-3992483b2081.png)

**Comment by sandrobonazzola** [comment]
Thanks @srbala ! #57 should solve

**Comment by srbala** [comment]
@sandrobonazzola Grub boot issue resolved, grub menu shows correctly and boots as expected.

Few points of observations: Testing ISO in VMWare, 4cpus, 9G RAM, 200G HDD

- One service failure (Configure Image Layers due to FileNotFoundError: [Errno 2] No such file or directory: '/usr/share/factory/etc/iscsi/initiatorname.iscsi'
- Installer interface - only root user to create/update password - is it by design? It would be nice to have non-root user, which can be oVirt admin
- It would be nice to show the IP address link on boot instead of `localhost` link, which shows after login

![image](https://user-images.githubusercontent.com/1273137/173080960-ae7f848b-2f9a-40b7-a284-f0ca61f7b44e.png)

![image](https://user-images.githubusercontent.com/1273137/173081592-af128524-6698-4f56-9909-aba934004dee.png)

**Comment by sandrobonazzola** [comment]
Reported the missing file issue on imgbased project.

As for the additional user, oVirt Node is not expected to be administered by anyone else than oVirt Engine itself, so no additional user is being created. About the localhost at first prompt, I would consider it cosmetic.

**Comment by sandrobonazzola** [comment]
I think now the leftovers are:
- Ensure that an Almalinux based node won't update into a CentOS Stream node and the other way around
- Get AlmaLinux build running somewhere and artifacts published somewhere

**Comment by lveyde** [comment]
@srbala what help (if any) is required on your side to advance this project?

**Comment by czadikem** [comment]
Yes I would love to use ovirt with almalinux.  So what needs to happen to help change or at least have this option available?

**Comment by lveyde** [comment]
> Yes I would love to use ovirt with almalinux. So what needs to happen to help change or at least have this option available?

Just to avoid confusion - I think that it's possible to use oVirt with AlmaLinux already.

This issue is about creating a native AlmaLinux based node.
Please note the last comment by @sandrobonazzola that listed the last few things we need for that to happen.

**Comment by gocallag** [comment]
Once https://github.com/oVirt/ovirt-node-ng-image/issues/147 has been closed, this ticket can be closed. Alma9 based ovirt-node-images can be built using the instructions at https://github.com/oVirt/ovirt-node-ng-image/blob/master/DEVELOPMENT.md
Once #147  closes that means the builds are being uploaded. The builds work, just the upload process is broken atm.

**Comment by dupondje** [comment]
Alma builds are now available!
